apisix/utils/log-util.lua

```diff
@@ -203,4 +200,55 @@ function _M.latency_details_in_ms(ctx)
     return latency, upstream_latency, apisix_latency
 end
 
+
+function _M.check_log_scheme(conf)
+    if conf.include_req_body_expr then
+        local ok, err = expr.new(conf.include_req_body_expr)
+        if not ok then
+            return nil,
+            {error_msg = "failed to validate the 'include_req_body_expr' expression: " .. err}
+        end
+    end
+    if conf.include_resp_body_expr then
+        local ok, err = expr.new(conf.include_resp_body_expr)
+        if not ok then
+            return nil,
+            {error_msg = "failed to validate the 'include_resp_body_expr' expression: " .. err}
+        end
+    end
+    return true, nil
+end
+
+
+function _M.collect_body(conf, ctx)
+    if conf.include_resp_body then
+        local log_response_body = true
+
+        if not conf.include_resp_body_expr then
```

Comment:
```suggestion
        if conf.include_resp_body_expr then
```

We should keep the result to avoid reevaluating the expr.